api/messages/publish.proto

```diff
@@ -0,0 +1,79 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+syntax = "proto3";
+
+option go_package = "github.com/elastic/elastic-agent-shipper/api/messages";
+package elastic.agent.shipper.v1.messages;
+
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+import "messages/struct.proto";
+
+message PublishRequest {
+ repeated Event events = 1;
+}
+
+// Event is a translation of beat.Event into protobuf.
+message Event {
+ // Creation timestamp of the event.
+ google.protobuf.Timestamp timestamp = 1;
+ // Source of the generated event.
+ Source source = 2;
+ // Optional. Producers may specify a unique event ID if the event has a natural unique
+ // identifier to simplify acknowledgement tracking. If no such identifier exists, the queue
+ // offset can be used for acknowledgement tracking.
+ string event_id = 3;
+ // Data stream for the event.
+ DataStream data_stream = 4;
+ // Metadata JSON object (map[string]google.protobuf.Value)
+ messages.Struct metadata = 5;
+ // Field JSON object (map[string]google.protobuf.Value)
+ messages.Struct fields = 6;
```

Comment:
```suggestion
 messages.Struct data = 6;
```

Since `Struct` has a `fields` field, this will read as `Event.fields.fields`.

We can try `Event.data.fields`, or `Event.payload.fields`, or something like that instead.

Reply:
To avoid confusion, I'd keep Event.Fields (because this is how it's called in Beats https://github.com/elastic/beats/blob/fb49f9cc13273c155dd9eaf2370b097450d7d8e5/libbeat/beat/event.go#L44) but I will rename `Fields`->`Data` in `Struct`.